devnet-sdk/devstack/devtest/gate.go

```diff
@@ -0,0 +1,25 @@
+package devtest
+
+import "github.com/stretchr/testify/require"
+
+// gateAdapter implements the require.TestingT interface by using skips instead of errors.
+// This is used to program test-gate conditions easily.
+type gateAdapter struct {
+	inner interface {
+		Helper()
+		Skipf(format string, args ...any)
+		SkipNow()
+	}
+}
+
+var _ require.TestingT = (*gateAdapter)(nil)
+
+func (g *gateAdapter) Errorf(format string, args ...interface{}) {
+	g.inner.Helper()
+	g.inner.Skipf(format, args...)
+}
+
+func (g *gateAdapter) FailNow() {
+	g.inner.Helper()
+	g.inner.SkipNow()
```

Comment:
In the current basicSystemTestHelper failing preconditions can be configured to either skip or fail. I think the idea is that if we explicitly configure a test to run as part of an op-acceptor suite, it should fail if it's not applicable since we claimed it was an important thing to check but then couldn't.

Not really sure if we need to support that here or not and if we do we could add it later, but figured it was worth mentioning.

Reply:
Good point. I'll add an env var for this

Reply:
I looked into it, and op-acceptor already defined an env var for this. Reusing that one for compatibility :+1: